src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -1614,4 +1649,23 @@ public static String generateJWTToken(
             privateKey, privateKeyFile, privateKeyFilePwd, accountName, userName);
     return s.issueJwtToken();
   }
+
+  /**
+   * Helper method to check if the request path is a login/auth request to use for retry strategy.
+   *
+   * @param request the post request
+   * @return true if this is a login/auth request, false otherwise
+   */
+  public static boolean isLoginRequest(HttpRequestBase request) {
```

Comment:
Maybe change the function name to like isNewRetryStrategyRequest as the new strategy might expand to other requests in the future.